### PR TITLE
SN 1.5: Move over NarrativeService modules

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Basic C&P of the Narrative Service method used by the StaticNarrative into the SN repo.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X
- [ ] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Black and Flake8 on changed Python code
- [x] If appropriate, I have recompiled the app and added the updated `StaticNarrativeImpl.py`, `StaticNarrativeServer.py`, and `compile_report.json` to this PR.

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: lib/StaticNarrative/exporter/dynamic_service_client.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+"""Client for dealing with KBase dynamic services."""
```
Comment (ialarmedalien) [inline lib/StaticNarrative/exporter/dynamic_service_client.py:1]: This is all copy-n-paste

File: lib/StaticNarrative/exporter/objectswithsets.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                    "include_metadata": include_metadata,
+                    # TODO: implement infostruct returns!
+                    # "infostruct": 1,
+                }
+            ],
+        )
+
+        # report_path = os.path.join(outdir, "list_sets.json")
+        # with open(report_path, "w") as fout:
+        #     json.dump(set_ret, fout, indent=2, sort_keys=True)
```
[the remark this answers is not in the document]
Comment (ialarmedalien) [reply lib/StaticNarrative/exporter/objectswithsets.py:85]: Leaving this in for now as it's useful for debugging -- I will remove it before any of this gets merged to `main`

File: lib/StaticNarrative/exporter/workspace_list_objects_iterator.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+"""Class for fetching objects from the workspace."""
```
Comment (ialarmedalien) [inline lib/StaticNarrative/exporter/workspace_list_objects_iterator.py:1]: all C&P, no edits

File: lib/StaticNarrative/exporter/objectswithsets.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
+import json
+import os
+from typing import Any
```
Comment (ialarmedalien) [inline lib/StaticNarrative/exporter/objectswithsets.py:3]: a few minor edits in this file:
- remove unused input variables
- add in some code for debugging (the json dump code that's commented out)

File: test/mocks.py [hunk 5]
```diff
@@ -89,7 +89,7 @@ def mock_adapter(request):
                 result = [
                     {"url": "https://something.kbase.us/service/narrative_service_url"}
                 ]
-            elif method == "NarrativeService.list_objects_with_sets":
+            elif "list_objects_with_sets" in method:
```
Comment (ialarmedalien) [inline test/mocks.py:92]: no longer part of the NarrativeService - now done in this repo

**Review by briehl — APPROVED** [review]
👍

**Pull request merged**